src/crewai/crew.py

```diff
@@ -900,7 +900,22 @@ def copy(self):
         }
 
         cloned_agents = [agent.copy() for agent in self.agents]
-        cloned_tasks = [task.copy(cloned_agents) for task in self.tasks]
+
+        task_mapping = {}
+
+        cloned_tasks = []
+        for task in self.tasks:
+            cloned_task = task.copy(cloned_agents, task_mapping)
+            cloned_tasks.append(cloned_task)
+            task_mapping[task.key] = cloned_task
+
+        for cloned_task, original_task in zip(cloned_tasks, self.tasks):
+            if original_task.context:
```

Comment:
Iwould consider moving the second `for` loop into the task.copy() function, to solve the original problem and not just in `crew.copy()`.  
`task.copy()` already receives task mapping with all of the previous tasks already populated so it should have access to the reference of the the cloned tasks' context.